drivers/nrf24l01p_ng/nrf24l01p_ng_communication.c

```
@@ -0,0 +1,220 @@
+        _swap_bytes(dest, len);
+    }
+    return status;
+}
+
+uint8_t nrf24l01p_ng_write_tx_payload(const nrf24l01p_ng_t *dev,
+                                      const void *src, size_t len)
+{
+    uint8_t cmd = NRF24L01P_NG_CMD_W_TX_PAYLOAD;
+    uint8_t status;
+    if (len > 1) {
+        len = MIN(len, NRF24L01P_NG_MAX_PAYLOAD_WIDTH);
+        uint8_t lsrc[NRF24L01P_NG_MAX_PAYLOAD_WIDTH];
+        memcpy(lsrc, src, len);
+        _swap_bytes(lsrc, len);
+        status = spi_transfer_byte(SPI_BUS, SPI_PIN_CS, true, cmd);
+        spi_transfer_bytes(SPI_BUS, SPI_PIN_CS, false, lsrc, NULL, len);
+    }
+    else {
+        status = spi_transfer_byte(SPI_BUS, SPI_PIN_CS, true, cmd);
+        spi_transfer_bytes(SPI_BUS, SPI_PIN_CS, false, src, NULL, len);
+    }
+    return status;
+}
+
+uint8_t nrf24l01p_ng_flush_tx(const nrf24l01p_ng_t *dev)
+{
+    uint8_t cmd = NRF24L01P_NG_CMD_FLUSH_TX;
+    return spi_transfer_byte(SPI_BUS, SPI_PIN_CS, false, cmd);
+}
+
+uint8_t nrf24l01p_ng_flush_rx(const nrf24l01p_ng_t *dev)
+{
+    uint8_t cmd = NRF24L01P_NG_CMD_FLUSH_RX;
+    return spi_transfer_byte(SPI_BUS, SPI_PIN_CS, false, cmd);
+}
+
+uint8_t nrf24l01p_ng_reuse_tx_pl(const nrf24l01p_ng_t *dev)
+{
+    uint8_t cmd = NRF24L01P_NG_CMD_REUSE_TX_PL;
+    return spi_transfer_byte(SPI_BUS, SPI_PIN_CS, false, cmd);
+}
+
+uint8_t nrf24l01p_ng_read_rx_pl_width(const nrf24l01p_ng_t *dev, uint8_t *dest)
+{
+    uint8_t cmd = NRF24L01P_NG_CMD_R_RX_PL_WID;
+    uint8_t status = spi_transfer_byte(SPI_BUS, SPI_PIN_CS, true, cmd);
+    spi_transfer_bytes(SPI_BUS, SPI_PIN_CS, false, NULL, dest, 1);
+    return status;
+}
+
+uint8_t nrf24l01p_ng_write_ack_pl(const nrf24l01p_ng_t *dev, const void *src,
+                                  size_t len, uint8_t pipe)
+{
+    uint8_t cmd = NRF24L01P_NG_CMD_W_ACK_PAYLOAD(pipe);
+    uint8_t status;
+    if (len > 1) {
+        len = MIN(len, NRF24L01P_NG_MAX_PAYLOAD_WIDTH);
+        uint8_t lsrc[NRF24L01P_NG_MAX_PAYLOAD_WIDTH];
```

should too long payloads really be silently truncated?